compiler/rustc_errors/src/lib.rs

```diff
@@ -937,13 +942,23 @@ impl Handler {
         result
     }
 
-    /// Construct a builder at the `Error` level with the `msg`.
+    /// Construct a builder at the `Fatal` level with the `msg`.
     #[rustc_lint_diagnostics]
     #[track_caller]
     pub fn struct_fatal(&self, msg: impl Into<DiagnosticMessage>) -> DiagnosticBuilder<'_, !> {
         DiagnosticBuilder::new(self, Level::Fatal, msg)
     }
 
+    /// Construct a builder at the `Fatal` level with the `msg`, that doesn't abort.
+    #[rustc_lint_diagnostics]
+    #[track_caller]
+    pub fn struct_almost_fatal(
```

Comment:
Didn't you just remove another helper that was named `struct_fatal_no_abort` or something?

Reply:
Good memory! In [this commit](https://github.com/rust-lang/rust/pull/118587/commits/7811c976d1a6b014bff0fdada7aaaa1774d31b82) I removed this:
```rust
    // Note: unlike `Handler::fatal`, this doesn't return `!`, because that is
    // inappropriate for some of its call sites.
    fn fatal_no_raise(&mut self, msg: impl Into<DiagnosticMessage>) -> FatalError {
        self.emit_diagnostic(&mut Diagnostic::new(Fatal, msg));
        FatalError
    }
```
Which was a bit different to this one. Both steps towards making all these functions more consistent :)